### ISSUE TITLE
Note creation from selected text mangles non-vault files

### ISSUE THREAD

**Issue by whitbur** [dendronhq/dendron#410]
**Describe the bug**
Creating a new note via the lookup hotkey while text is selected in a non-vault file creates a new note with the selected contents in it, but deletes that content from and saves (!!) the non-vault file.

**To Reproduce**
1. Open any Dendron workspace
2. Open finder/explorer, navigate to a markdown file somewhere else on your computer, double click on it. It should open in the same VS Code window as your vault.
3. Select some text from the non-vault file, hit the lookup hotkey (Cmd + L), type a name, and hit enter.
4. The new note will be created with the selected content, the non-vault file will be truncated and saved.

**Expected behavior**
Dendron should not automatically modify or save content in open files **not** located in currently managed vaults. I'm delighted that the new note is pre-populated, let's keep that though 😉 

**Desktop info:**
 - OS: macOS Catalina
 - VS Code: 1.52.0
 - Dendron: 0.20.0

**Additional context**
I noticed this while individually importing some notes I had stored in a different folder. I was relying on the last modified time of those notes, so I was particularly frustrated that Dendron saved without my permission.

I can imagine this also being an unexpected issue if one wanted to copy a snippet of code from a local repo, sections of log files, etc.

**Dendron Log file**
<details><summary>Click here to expand</summary>

```
{"level":30,"time":1608252967626,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"WindowWatcher:onDidChangeActiveTextEditor","msg":"enter","uri":{"$mid":1,"fsPath":"/Users/whitbur/notes/current.md","external":"file:///Users/whitbur/notes/current.md","path":"/Users/whitbur/notes/current.md","scheme":"file"}}
{"level":30,"time":1608252974823,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"c:run","msg":"pre-execute"}
{"level":30,"time":1608252974823,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"LookupCommand:execute","opts":{"flavor":"note"},"msg":"enter"}
{"level":30,"time":1608252974824,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"LookupCommand:execute","profile":0,"msg":"postCreateController"}
{"level":30,"time":1608252974824,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"LookupCommand:execute","profile":1,"msg":"post:extractRange"}
{"level":30,"time":1608252974825,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"LookupControllerV2:show","msg":"enter","cleanOpts":{"document":{"uri":{"$mid":1,"fsPath":"/Users/whitbur/notes/current.md","external":"file:///Users/whitbur/notes/current.md","path":"/Users/whitbur/notes/current.md","scheme":"file"},"fileName":"/Users/whitbur/notes/current.md","isUntitled":false,"languageId":"markdown","version":1,"isClosed":false,"isDirty":false,"eol":2,"lineCount":9},"range":{"start":{"line":5,"character":0},"end":{"line":5,"character":33},"active":{"line":5,"character":33},"anchor":{"line":5,"character":0}},"ignoreFocusOut":true}}
{"level":30,"time":1608252974826,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"LookupControllerV2:show","profile":1,"msg":"post:createQuickPick"}
{"level":30,"time":1608252974826,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"LookupControllerV2:show","profile":1,"msg":"post:createProvider"}
{"level":30,"time":1608252974826,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"updatePickerBehavior","activeButtons":[{"iconPathNormal":{"id":"find-selection"},"iconPathPressed":{"id":"menu-selection"},"type":"selectionExtract","pressed":true,"title":"Selection Extract","canToggle":true,"opts":{"title":"Selection Extract","iconOff":"find-selection","iconOn":"menu-selection","type":"selectionExtract"}}]}
{"level":30,"time":1608252974827,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"LookupCommand:execute","profile":3,"msg":"post:show"}
{"level":30,"time":1608252974827,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"updatePickerItems","msg":"enter","queryOrig":"current","source":"updatePickerBehavior:normal"}
{"level":30,"time":1608252974845,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"updatePickerItems","msg":"exit","queryOrig":"current","source":"updatePickerBehavior:normal","profile":17,"cancelled":false}
{"level":30,"time":1608252974845,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"LookupControllerV2:show","profile":1,"msg":"post:updatePickerBehavior"}
{"level":30,"time":1608252974846,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"LookupControllerV2:show","profile":1,"msg":"post:provide"}
{"level":30,"time":1608252974846,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"LookupControllerV2:show","profile":1,"msg":"exit"}
{"level":30,"time":1608252974846,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"c:run","msg":"post-execute"}
{"level":30,"time":1608252976827,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"updatePickerItems","msg":"enter","queryOrig":"t","source":"onValueChange"}
{"level":30,"time":1608252976827,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"createPickerItemsFromEngine","msg":"first query"}
{"level":30,"time":1608252976830,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"createPickerItemsFromEngine","msg":"post:queryNotes"}
{"level":30,"time":1608252976835,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"createPickerItemsFromEngine","msg":"engine.query","profile":7}
{"level":30,"time":1608252976835,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"updatePickerItems","msg":"exit","queryOrig":"t","source":"onValueChange","profile":8,"cancelled":false}
{"level":30,"time":1608252976888,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"updatePickerItems","msg":"enter","queryOrig":"te","source":"onValueChange"}
{"level":30,"time":1608252976889,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"createPickerItemsFromEngine","msg":"first query"}
{"level":30,"time":1608252976890,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"createPickerItemsFromEngine","msg":"post:queryNotes"}
{"level":30,"time":1608252976928,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"createPickerItemsFromEngine","msg":"engine.query","profile":38}
{"level":30,"time":1608252976928,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"updatePickerItems","msg":"exit","queryOrig":"te","source":"onValueChange","profile":39,"cancelled":false}
{"level":30,"time":1608252976977,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"updatePickerItems","msg":"enter","queryOrig":"tes","source":"onValueChange"}
{"level":30,"time":1608252976978,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"createPickerItemsFromEngine","msg":"first query"}
{"level":30,"time":1608252976979,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"createPickerItemsFromEngine","msg":"post:queryNotes"}
{"level":30,"time":1608252977012,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"createPickerItemsFromEngine","msg":"engine.query","profile":34}
{"level":30,"time":1608252977012,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"updatePickerItems","msg":"exit","queryOrig":"tes","source":"onValueChange","profile":35,"cancelled":false}
{"level":30,"time":1608252977041,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"updatePickerItems","msg":"enter","queryOrig":"test","source":"onValueChange"}
{"level":30,"time":1608252977041,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"createPickerItemsFromEngine","msg":"first query"}
{"level":30,"time":1608252977043,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"createPickerItemsFromEngine","msg":"post:queryNotes"}
{"level":30,"time":1608252977045,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"createPickerItemsFromEngine","msg":"engine.query","profile":3}
{"level":30,"time":1608252977046,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"updatePickerItems","msg":"exit","queryOrig":"test","source":"onValueChange","profile":5,"cancelled":false}
{"level":30,"time":1608252978948,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"updatePickerItems","msg":"enter","queryOrig":"test.","source":"onValueChange"}
{"level":30,"time":1608252978948,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"createPickerItemsFromEngine","msg":"first query"}
{"level":30,"time":1608252978949,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"createPickerItemsFromEngine","msg":"post:queryNotes"}
{"level":30,"time":1608252978951,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"createPickerItemsFromEngine","msg":"engine.query","profile":2}
{"level":30,"time":1608252978951,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"updatePickerItems","msg":"exit","queryOrig":"test.","source":"onValueChange","profile":3,"cancelled":false}
{"level":30,"time":1608252979205,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"updatePickerItems","msg":"enter","queryOrig":"test.n","source":"onValueChange"}
{"level":30,"time":1608252979205,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"createPickerItemsFromEngine","msg":"first query"}
{"level":30,"time":1608252979206,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"createPickerItemsFromEngine","msg":"post:queryNotes"}
{"level":30,"time":1608252979208,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"createPickerItemsFromEngine","msg":"engine.query","profile":2}
{"level":30,"time":1608252979208,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"updatePickerItems","msg":"exit","queryOrig":"test.n","source":"onValueChange","profile":3,"cancelled":false}
{"level":30,"time":1608252979508,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"updatePickerItems","msg":"enter","queryOrig":"test.no","source":"onValueChange"}
{"level":30,"time":1608252979508,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"createPickerItemsFromEngine","msg":"first query"}
{"level":30,"time":1608252979509,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"createPickerItemsFromEngine","msg":"post:queryNotes"}
{"level":30,"time":1608252979534,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"createPickerItemsFromEngine","msg":"engine.query","profile":26}
{"level":30,"time":1608252979534,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"updatePickerItems","msg":"exit","queryOrig":"test.no","source":"onValueChange","profile":26,"cancelled":false}
{"level":30,"time":1608252979876,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"updatePickerItems","msg":"enter","queryOrig":"test.n","source":"onValueChange"}
{"level":30,"time":1608252979877,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"createPickerItemsFromEngine","msg":"first query"}
{"level":30,"time":1608252979878,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"createPickerItemsFromEngine","msg":"post:queryNotes"}
{"level":30,"time":1608252979879,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"createPickerItemsFromEngine","msg":"engine.query","profile":2}
{"level":30,"time":1608252979879,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"updatePickerItems","msg":"exit","queryOrig":"test.n","source":"onValueChange","profile":2,"cancelled":false}
{"level":30,"time":1608252980009,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"updatePickerItems","msg":"enter","queryOrig":"test.","source":"onValueChange"}
{"level":30,"time":1608252980009,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"createPickerItemsFromEngine","msg":"first query"}
{"level":30,"time":1608252980010,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"createPickerItemsFromEngine","msg":"post:queryNotes"}
{"level":30,"time":1608252980011,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"createPickerItemsFromEngine","msg":"engine.query","profile":2}
{"level":30,"time":1608252980011,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"updatePickerItems","msg":"exit","queryOrig":"test.","source":"onValueChange","profile":2,"cancelled":false}
{"level":30,"time":1608252981256,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"updatePickerItems","msg":"enter","queryOrig":"test.i","source":"onValueChange"}
{"level":30,"time":1608252981256,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"createPickerItemsFromEngine","msg":"first query"}
{"level":30,"time":1608252981257,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"createPickerItemsFromEngine","msg":"post:queryNotes"}
{"level":30,"time":1608252981259,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"createPickerItemsFromEngine","msg":"engine.query","profile":2}
{"level":30,"time":1608252981259,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"updatePickerItems","msg":"exit","queryOrig":"test.i","source":"onValueChange","profile":3,"cancelled":false}
{"level":30,"time":1608252981349,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"updatePickerItems","msg":"enter","queryOrig":"test.im","source":"onValueChange"}
{"level":30,"time":1608252981349,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"createPickerItemsFromEngine","msg":"first query"}
{"level":30,"time":1608252981351,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"createPickerItemsFromEngine","msg":"post:queryNotes"}
{"level":30,"time":1608252981352,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"createPickerItemsFromEngine","msg":"engine.query","profile":2}
{"level":30,"time":1608252981352,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"updatePickerItems","msg":"exit","queryOrig":"test.im","source":"onValueChange","profile":3,"cancelled":false}
{"level":30,"time":1608252981477,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"updatePickerItems","msg":"enter","queryOrig":"test.imp","source":"onValueChange"}
{"level":30,"time":1608252981478,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"createPickerItemsFromEngine","msg":"first query"}
{"level":30,"time":1608252981479,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"createPickerItemsFromEngine","msg":"post:queryNotes"}
{"level":30,"time":1608252981480,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"createPickerItemsFromEngine","msg":"engine.query","profile":2}
{"level":30,"time":1608252981480,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"updatePickerItems","msg":"exit","queryOrig":"test.imp","source":"onValueChange","profile":2,"cancelled":false}
{"level":30,"time":1608252981657,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"updatePickerItems","msg":"enter","queryOrig":"test.impo","source":"onValueChange"}
{"level":30,"time":1608252981657,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"createPickerItemsFromEngine","msg":"first query"}
{"level":30,"time":1608252981658,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"createPickerItemsFromEngine","msg":"post:queryNotes"}
{"level":30,"time":1608252981659,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"createPickerItemsFromEngine","msg":"engine.query","profile":2}
{"level":30,"time":1608252981659,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"updatePickerItems","msg":"exit","queryOrig":"test.impo","source":"onValueChange","profile":2,"cancelled":false}
{"level":30,"time":1608252981735,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"updatePickerItems","msg":"enter","queryOrig":"test.impor","source":"onValueChange"}
{"level":30,"time":1608252981736,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"createPickerItemsFromEngine","msg":"first query"}
{"level":30,"time":1608252981737,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"createPickerItemsFromEngine","msg":"post:queryNotes"}
{"level":30,"time":1608252981738,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"createPickerItemsFromEngine","msg":"engine.query","profile":2}
{"level":30,"time":1608252981738,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"updatePickerItems","msg":"exit","queryOrig":"test.impor","source":"onValueChange","profile":2,"cancelled":false}
{"level":30,"time":1608252981887,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"updatePickerItems","msg":"enter","queryOrig":"test.import","source":"onValueChange"}
{"level":30,"time":1608252981887,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"createPickerItemsFromEngine","msg":"first query"}
{"level":30,"time":1608252981889,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"createPickerItemsFromEngine","msg":"post:queryNotes"}
{"level":30,"time":1608252981890,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"createPickerItemsFromEngine","msg":"engine.query","profile":2}
{"level":30,"time":1608252981890,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"updatePickerItems","msg":"exit","queryOrig":"test.import","source":"onValueChange","profile":3,"cancelled":false}
{"level":30,"time":1608252982924,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"onDidAcceptSingle","msg":"enter","value":"test.import","opts":{"flavor":"note"},"selectedItems":[{"fname":"Create New","id":"46a3cfe3-57d6-49bc-9f6b-0a3ee0fe7c2a"}],"activeItems":[{"fname":"Create New","id":"46a3cfe3-57d6-49bc-9f6b-0a3ee0fe7c2a"}]}
{"level":30,"time":1608252982925,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"onAcceptNewNode"}
{"level":30,"time":1608252982926,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"onAcceptNewNode","msg":"createNewPick","value":"test.import"}
{"level":30,"time":1608252982926,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"onAcceptNewNode","msg":"create normal node"}
{"level":30,"time":1608252982927,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"onAcceptNewNode","msg":"post:maybeAddSchema"}
{"level":30,"time":1608252982927,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"onAcceptNewNode","msg":"pre:checkNoteExist"}
{"level":30,"time":1608252982928,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"onAcceptNewNode","msg":"pre:pickerOnCreate"}
{"level":30,"time":1608252982953,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"onAcceptNewNode","msg":"pre:engine.write","uri":{"$mid":1,"path":"/Users/whitbur/Dendron/vault/test.import.md","scheme":"file"}}
{"level":30,"time":1608252982968,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"onAcceptNewNode","msg":"engine.write"}
{"level":30,"time":1608252983013,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"WindowWatcher:onDidChangeActiveTextEditor","msg":"enter","uri":{"$mid":1,"fsPath":"/Users/whitbur/Dendron/vault/test.import.md","external":"file:///Users/whitbur/Dendron/vault/test.import.md","path":"/Users/whitbur/Dendron/vault/test.import.md","scheme":"file"}}
{"level":30,"time":1608252983015,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"WorkspaceWatcher:onWillSaveTextDocument","url":"/Users/whitbur/notes/current.md","reason":3,"msg":"enter"}
{"level":30,"time":1608252983023,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"showDocAndHidePicker","msg":"showTextDocument","fsPath":"/Users/whitbur/Dendron/vault/test.import.md"}
{"level":30,"time":1608252983163,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"VaultWatcher:onDidCreate","uri":{"$mid":1,"fsPath":"/Users/whitbur/Dendron/vault/test.import.md","path":"/Users/whitbur/Dendron/vault/test.import.md","scheme":"file"}}
{"level":30,"time":1608252983264,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"refreshTree"}
{"level":30,"time":1608252986687,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"c:run","msg":"pre-execute"}
{"level":30,"time":1608252986687,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"execute","opts":{}}
{"level":30,"time":1608252986688,"pid":19666,"hostname":"147dda11b1b4.ant.amazon.com","name":"dendron","ctx":"c:run","msg":"post-execute"}
```

</details>

**Comment by kevinslin** [comment]
Thanks for the issue. Will create a task for it in our backlog

**Comment by kevinslin** [comment]
Changes:

- make a change to check if document comes from a vault
    - if not, don't delete from original document
- relevant code
    - [text extraction code](https://github.com/dendronhq/dendron/blob/master/packages/plugin-core/src/components/lookup/LookupControllerV2.ts#L379:L379) 
    - [check if doc comes from known vault](https://github.com/dendronhq/dendron/blob/master/packages/common-all/src/vault.ts#L28:L28)

**Comment by hikchoi** [comment]
Anyone working on this?
Otherwise I'll pick this up.

**Comment by kevinslin** [comment]
Noticed that the original vault code is deprecated. This is the [function](https://github.com/dendronhq/dendron/blob/a1044314b254c7dc45b8ceed4533e8118552320d/packages/common-all/src/vault.ts#L71) you should use to check for the path

**Comment by hikchoi** [comment]
> Noticed that the original vault code is deprecated. This is the [function](https://github.com/dendronhq/dendron/blob/a1044314b254c7dc45b8ceed4533e8118552320d/packages/common-all/src/vault.ts#L71) you should use to check for the path

I had to go back to the old commit and figure out which line you were referring to 😅  I did figure out that I need to use `getVaultByNotePathV4` instead.
It would be great if Github links to the original line rather than just point to the literal line number 🙃 

I'm currently playing around with the plugin tests to figure out how they are set up.
Looks like there aren't any tests for `selectExtraction` yet?
